test/integ/quote-classic.test.ts

```diff
@@ -1987,6 +1990,113 @@ describe('quote', function () {
+                it(`fee-on-transfer ${tokenIn.symbol} -> ${tokenOut.symbol}`, async () => {
+                  // we want to swap the tokenIn/tokenOut order so that we can test both sellFeeBps and buyFeeBps for exactIn vs exactOut
+                  const originalAmount = tokenIn.equals(WETH9[ChainId.MAINNET]!) ? '10' : '2924';
+                  const amount = await getAmountFromToken(type, tokenIn, tokenOut, originalAmount);
+                  const simulateFromAddress = tokenIn.equals(WETH9[ChainId.MAINNET]!)
+                    ? '0x2fEb1512183545f48f6b9C5b4EbfCaF49CfCa6F3'
+                    : '0x171d311eAcd2206d21Cb462d661C33F0eddadC03';
+                  const quoteReq: QuoteRequestBodyJSON = {
+                    requestId: 'id',
+                    tokenIn: tokenIn.address,
+                    tokenInChainId: tokenIn.chainId,
+                    tokenOut: tokenOut.address,
+                    tokenOutChainId: tokenOut.chainId,
+                    amount: amount,
+                    type,
+                    // we have to use large slippage for FOT swap, because URA always forks at the latest block,
+                    // and the FOT swap can have large slippage, despite SOR already subtracted FOT tax
+                    slippageTolerance: LARGE_SLIPPAGE,
+                    configs: [
+                      {
+                        routingType: RoutingType.CLASSIC,
+                        recipient: alice.address,
+                        deadline: 360,
+                        simulateFromAddress: simulateFromAddress,
+                        // we already know that SOR only supports FOT in v2 as of now
+                        // so we can send v2 only to send some test runtime
+                        protocols: ['v2'],
+                        algorithm,
+                        enableUniversalRouter: true,
+                        enableFeeOnTransferFeeFetching: true,
+                      },
+                    ],
+                  };
+
+                  const response = await call(quoteReq);
+                  const { data, status } = response;
+                  const quoteJSON = data.quote as ClassicQuoteDataJSON;
+
+                  expect(status).to.equal(200);
+                  expect(quoteJSON.simulationError).to.equal(false);
+                  expect(quoteJSON.simulationStatus).to.equal('SUCCESS');
+                  expect(quoteJSON.methodParameters).to.not.be.undefined;
+
+                  for (const r of quoteJSON.route) {
+                    for (const pool of r) {
+                      expect(pool.type).equal('v2-pool');
+                      const v2Pool = pool as V2PoolInRouteJSON;
+
+                      if (v2Pool.tokenIn.address === BULLET_WHT_TAX.address) {
+                        expect(v2Pool.tokenIn.sellFeeBps).to.be.not.undefined;
```

Comment:
can we just check the value of the bps here instead of definedness? that would be more explicit

Reply:
Good point. Since we are checking the bps values like `expect(v2Pool.tokenIn.sellFeeBps).to.be.equals(BULLET_WHT_TAX.sellFeeBps?.toString());`, checking definedness seems redundant and implicit.